### PR TITLE
sdl12-compat 1.2.52 (new formula)

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

**Review by Bo98 — COMMENTED** [review]
Quite like the potential of using this if we ever want to get rid of the sdl1 formula.

File: Formula/sdl12-compat.rb
Comment (Bo98) [inline Formula/sdl12-compat.rb:17]: Can you elaborate why you chose these.

Why do we need any of the test programs (`SDL12TESTS`)?
Why should we not include SDL headers (`SDL12DEVEL`), particularly given this conflicts with sdl1?

Comment (pthariensflame) [reply Formula/sdl12-compat.rb:17]: > Can you elaborate why you chose these?
> 
> Why do we need any of the test programs (`SDL12TESTS`)? Why should we not include SDL headers (`SDL12DEVEL`), particularly given this conflicts with sdl1?

Both are on by default; I don't recall why I opted for tests but I know why I turned off "devel": I didn't know that's what it was.  I assumed it was something only for maintainers to have on.  I'll flip them both.

```suggestion
           "-DSDL12DEVEL=1", "-DSDL12TESTS=0"
```

**Comment by Bo98** [comment]
Looks good in genereal, though tests could have more.

Tested this out locally, and I reckon that there should be a symlink from sdl.pc to sdl12_compat.pc. I'm not entirely sure why upstream don't do that, but it's likely something we'll need if we want this to be a drop in replacement for `sdl`. Arch and Debian seem to agree as they install a symlink.

**Comment by pthariensflame** [comment]
What should we do about the "notability" audit failure?

**Comment by Bo98** [comment]
If this is something we'd like to use to replace SDL 1.2, then that would be a good reason to ignore that audit.

Otherwise, it's nearly 75 stars already anyway.

**Comment by pthariensflame** [comment]
Is this ready?

File: Formula/sdl.rb
Comment (carlocab) [inline Formula/sdl.rb:70]: We could probably add the `conflicts_with` to both formulae in a separate syntax-only PR to avoid having to rebuild `sdl`.

Comment (pthariensflame) [reply Formula/sdl.rb:70]: I'm not sure how to do that; I'm not great with advanced git.

Comment (carlocab) [reply Formula/sdl.rb:70]: Just remove the `conflicts_with` from both formulae and then squash all your commits here together.

File: Formula/sdl12-compat.rb
Comment (carlocab) [inline Formula/sdl12-compat.rb:17]:
```suggestion
    system "cmake", "-S", ".", "-B", "build",
                    "-DSDL2_PATH=#{Formula["sdl2"].opt_prefix}",
                    "-DSDL12DEVEL=1",
                    "-DSDL12TESTS=0",
                    *std_cmake_args
```

File: Formula/sdl12-compat.rb
Comment (carlocab) [inline Formula/sdl12-compat.rb:20]:
```suggestion
    (lib/"pkgconfig").install_symlink "sdl12_compat.pc" => "sdl.pc"
```

File: Formula/sdl12-compat.rb
Comment (carlocab) [inline Formula/sdl12-compat.rb:27]:
```suggestion
    assert_predicate lib/shared_library("libSDL"), :exist?
    assert_predicate lib/shared_library("libSDL-1.2.0"), :exist?
```
This might work

File: Formula/sdl12-compat.rb
Comment (Bo98) [reply Formula/sdl12-compat.rb:17]: I generally also prefer `ON` and `OFF` to 1 and 0 but it doesn't really matter much.

Comment (carlocab) [reply Formula/sdl12-compat.rb:17]: I was actually thinking the same thing. Suggestion amended.

File: Formula/sdl12-compat.rb
Comment (pthariensflame) [reply Formula/sdl12-compat.rb:27]: I'll try it!

**Comment by pthariensflame** [comment]
@carlocab @Bo98 #96356 is now opened and its content removed from this one.

**Review by carlocab — APPROVED** [review]
#96356 will need a rebase and the second commit split into two separate commits in the same PR, but we can work on that after this is merged. Thanks @pthariensflame!

**Pull request closed without merging**
